pkgs/servers/web-apps/lemmy/ui.nix

```diff
@@ -0,0 +1,82 @@
+{ lib
+, mkYarnPackage
+, libsass
+, nodejs
+, python3
+, pkg-config
+, writeShellScriptBin
+, jq
+, runCommand
+, fetchFromGitHub
+}:
+
+let
+  pkgConfig = {
+    node-sass = {
+      nativeBuildInputs = [ ];
+      buildInputs = [ libsass pkg-config python3 ];
+      postInstall = ''
+        LIBSASS_EXT=auto yarn --offline run build
+        rm build/config.gypi
+      '';
+
+      dontFixup = false;
+
+    };
+  };
+
+  pname = "lemmy-ui";
+  version = "0.12.2";
+  src = fetchFromGitHub {
+    owner = "LemmyNet";
+    repo = pname;
+    rev = "${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-iFLJqUnz4m9/JTSaJSUugzY5KkiKtH0sMYY4ALm2Ebk=";
+  };
+
+  patchedPackageJSON = runCommand "package.json" { } ''
+    ${jq}/bin/jq '.version = "${version}"' ${src}/package.json > $out
+  '';
```

Comment:
this could probably be done in the postPatch phase

Reply:
actually, we should probably have upstream just include their version... it would also fix the eval issue

Reply:
I've opened an issue upstream regarding this https://github.com/LemmyNet/lemmy-ui/issues/411
What we should do is try to open a PR upstream about this, and use fetchPatch to fetch the PR change.

Reply:
problem with that is that the packageJSON just takes the path from the src without patches applied.

Reply:
got it, thanks for your help on this!

Reply:
upstream now includes their version, we can probably make an update on an unstable version to include this patch. (I believe there were a couple of other updates since the last version).
It would only be temporary of course until they release a new version.